docs/rsdl/rapid-rsdl-capabilities.md

```diff
@@ -0,0 +1,337 @@
+
+The format is a sequence of properties that can be expanded.
+
+```rsdl
+path /orders {
+    GET { 
+        expand {
+            items
+        }
+    }
+}
+```
+
+ The expand capability introduces a nesting of capabilities since the type of the expandable property can itself be used in select, filter, and expand.
+
+```rsdl
+
+
+type Order {
+
+}
+
+capability Order defaultOrder {
+    filter {
+        eq { id name }
+        prefix { nam description }
+        range { createdDateTime }
+    }
+    expand {
+        items { 
+            filter { 
+                eq { id name } 
+                prefix { name }
+            }
+            expand { 
+                sku
+            }
+        }    
+    }
+    paging
+    count
+}
+
+capability Order limitedOrder {   
+    filter {
+        eq { id name }
+    } 
+}
+
+path /orders {
+    GET { 
+       defaultOrder
+    }
+}
+
+path /customers/{id}/orders {
+    GET { 
+        limitedOrder
+    }
+}
+```
+
+- `/order?$expand=items`
+- `/order?$expand=items(expand=sku)`
+- `/order?$expand=items(expand=sku; filter=id eq '100')`
+- `/order?$expand=items(select=id; expand=sku; filter=id eq '100')`
+
+
+### Paging capabilities
+
+The paging capability allows to specify  if a request returning a collection can specify $top and $skip query parameters.
+
+```rsdl
+path /orders {
+    GET { 
+        paging
+    }
+}
+```
+
+The paging capability has no parameters itself. If a service uses a different way to implement paging, please refer to the [extensibility](#extensibility) section below
+
+The paging capability is typically seen in a GET capability but can also be nested in an expand capability.
+
+### Count capabilities
+
+The count capability allows to specify if a request returning a collection can specify $count query parameter.
+
+```rsdl
+path /orders {
+    GET { 
+        count
+    }
+}
+```
+
+## Extensibility
+
+The RSDL API capabilities allow to specify custom capabilities, so called traits, for the service to implement that are outside the scope of the patterns known by the RSDL specification. 
+
+One example for these could be the handling of requests when throttling is needed, where the service responds with certain headers for requests that cross a throttling threshold. There are multiple established pattern (protocol) to implement this but it is not covered in RSDL directly. But it can be specified in RSDL as an extension and the service implementation can read it and configure the right behavior based on the traits in RSDL.
+
+The individual trait is treated as simple opaque (meta-) data that is passed on the service in the form of a trait name and a list of key-value pairs. The implementation is then free to interpret the presence of the trait and the parameters as it sees fit.
+
+```rsdl
+path /orders {
+    GET { 
+        traits { 
+            longRunningOperation
+            topWithoutSkip
+            throttling { level: "premium\u12a4" }
```

Comment:
Are the values always strings, or can they also be arrays, objects, numbers, booleans, i.e. "embedded JSON"?